src/targets/gpu/compile_hip.cpp

```diff
@@ -194,32 +194,32 @@ struct hiprtc_program
 };
 
 std::vector<std::vector<char>> compile_hip_src_with_hiprtc(std::vector<hiprtc_src_file> srcs,
-                                                           const std::string& params,
+                                                           const std::vector<std::string>& params,
                                                            const std::string& arch)
 {
     hiprtc_program prog(std::move(srcs));
-    auto options = split_string(params, ' ');
-    options.push_back("-DMIGRAPHX_USE_HIPRTC=1");
+    auto options = params;
+    options.emplace_back("-DMIGRAPHX_USE_HIPRTC=1");
     // remove following three compilation flags for HIPRTC once fixes from hipRTC are available in
     if(enabled(MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS{}))
     {
-        options.push_back("-DMIGRAPHX_HAS_DPP=0");
-        options.push_back("-DMIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS=1");
-        options.push_back("-Wno-reserved-identifier");
-        options.push_back("-Wno-unused-parameter");
-        options.push_back("-Wno-gnu-line-marker");
-        options.push_back("-Wno-old-style-cast");
+        options.emplace_back("-DMIGRAPHX_HAS_DPP=0");
+        options.emplace_back("-DMIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS=1");
+        options.emplace_back("-Wno-reserved-identifier");
+        options.emplace_back("-Wno-unused-parameter");
+        options.emplace_back("-Wno-gnu-line-marker");
+        options.emplace_back("-Wno-old-style-cast");
```

Comment:
There's a lot of changes like this that are unrelated to updating the process class. Can you put these unrelated changes in seperate PR? It will make it easier to review.